docs/extend/server-schemas.md

```diff
@@ -0,0 +1,64 @@
+# Server component schemas
+
+Starting with Zowe v2.0, each Component in Zowe must contain a [json schema](https://json-schema.org/) describing the configuration parameters that are valid for its component section in Zowe's server configuration. If a component does not have anything that can be configured, this file can just be boilerplate specifying that it fully inherits generic Component parameters and nothing more.
+
+The server infrastructure will utilize each components' schema files to validate a Zowe instance configuration every startup, so this requirement is enforced by code.
+
+## Requirements
+
+* Server component json schema files must follow the json schema [spec 2019-09](https://json-schema.org/draft/2019-09/schema).
+* Each component must state where its base schema file is located by the manifest parameter "schemas.config"
+* The schema file must use and/or extend the Zowe Component base schema by use of the "allOf" attribute.
+* The schema must have an `$id` property which is a URI that has a domain related to the entity that developed the Component.
+* The file should be tagged on z/OS but elsewhere must at least be encoded as ASCII-subset of UTF-8
```

Comment:
We don't have tagging requirement on any other files. Any particular reason this schema file must be tagged?

Reply:
doesn't need to be, just should be ascii at least. just the way the configmgr reads files.